crypto/x509/v3_san.c

```diff
@@ -393,11 +393,11 @@ static GENERAL_NAMES *v2i_subject_alt(X509V3_EXT_METHOD *method,
 
     for (i = 0; i < num; i++) {
         cnf = sk_CONF_VALUE_value(nval, i);
-        if (!ossl_v3_name_cmp(cnf->name, "email")
+        if (ossl_v3_name_cmp(cnf->name, "email") == 0
             && cnf->value && strcmp(cnf->value, "copy") == 0) {
             if (!copy_email(ctx, gens, 0))
                 goto err;
-        } else if (!ossl_v3_name_cmp(cnf->name, "email")
+        } else if (ossl_v3_name_cmp(cnf->name, "email") == 0
```

Comment:
I'm not really sure how changing the spelling of the return value check qualifies as "clarify subject/req parameter for constructing SAN email address from subject DN"

Reply:
Well, the commit message was focusing on the doc change and indeed was incomplete in the sense you pointed out. So I added:
```
Also slightly improve the style of the respective code in crypto/x509/v3_san.c.
```